### ISSUE TITLE
Broken pipe instead of error message from terraform

### ISSUE THREAD

**Issue by sknmi** [gruntwork-io/terragrunt#3376]
## Describe the bug

During import of non-existent object or wrong object or some other error there is only broken pipe instead for the actual error.

## Steps To Reproduce

Steps to reproduce the behavior, code snippets and examples which can be used to reproduce the issue.

```hcl
terragrunt import 'gitlab_group.subgroups["test_group"]' non-existent-group
14:41:54.914 ERROR  terraform invocation failed in xxxx error=[xxxx] exit status 1
14:41:55.209 ERROR  1 error occurred:
	* [xxxx] exit status 1
```
Or if I have error in module:
```
14:23:13.534 ERROR  terraform invocation failed in yyyy error=[yyyy] signal: broken pipe
14:23:13.774 ERROR  1 error occurred:
	* [yyyy] signal: broken pipe
```
while on error in the code terraform itself gives this:
```
│ Error: Missing argument separator
│
│   on .terraform/modules/gitlab_resources/main.tf line 635, in locals:
│  635:   subgroup_id_map = { for group in local.subgroups : "${group.parent}/${group.name}" => gitlab_group.subgroups["${group.parent}/${group.name}"].id if contains(local.existing_keys, local.desired_key}
│
│ A comma is required to separate each function argument from the next.
```

## Expected behavior

Terragrunt must output errors from terraform

## Versions

- Terragrunt version: 0.67.1
- OpenTofu/Terraform version: 1.5.7
- Environment details (Ubuntu 20.04, Windows 10, etc.): macos

**Comment by MickVanDuijn** [comment]
We're experiencing the same behavior. To add to what was reported, it seems to abort the Terraform command early, causing locks in DynamoDB to not be released under some scenario's. We've temporarily downgraded to `terragrunt version v0.66.9` to be able to see the full error, and to not leave dangling locks.

**Comment by levkohimins** [comment]
@sknmi, can please you provide the configuration to reproduce the issue?

**Comment by denis256** [comment]
Hi,
running Terragrunt with `--terragrunt-log-level debug` may also highlight why it is happening

**Comment by ohmer** [comment]
Same issue here, upgrading from 0.66.3 to 0.67.1.

**Comment by mcfedr** [comment]
downgraded to [v0.66.9](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.66.9) to work around this

**Comment by levkohimins** [comment]
@sknmi, Thanks for the example. The issue will be fixed soon.

**Comment by alandreasyans** [comment]
The issue started to happen from 0.67.0 version.
I've encountered a similar issue when the output resource isn't declared in the module but terragrunt isn't informing me about that.

**Comment by levkohimins** [comment]
We are working on a fix. I will let you know here once it is fixed.

**Comment by Argote-Gustavo-bah** [comment]
> It's also not working on `0.67.4` with opt-out flags `--terragrunt-disable-log-formatting --terragrunt-forward-tf-stdout` 😞 This effectively means that #3370 is a show-stopper issue.
> 
> While the initial issue is with code/env that raises execution error, the problem with Terragrunt is that those are raised to fatal/panic where message is not passed to the user and all post-exec stages are not triggered, including lock release (and it may even lead to resource leak if something is created in paid cloud and not destroyed...)
> 
> I can confirm that the downgrade to `0.66.9` was enough to address the issue, but it's hard with Homebrew.
> 
> In my code (which has some elements like #3372), the offending error should look like that (output from 0.66.9)
> 
> ```
> ╷
> │ Error: file provisioner error
> │ 
> │   with ...,
> │   on ... line 11, in resource "null_resource" "...":
> │   11:   provisioner "file" {
> │ 
> │ timeout - last error: dial tcp ...:22: i/o timeout
> ╵
> ```
> 
> but instead it now I get only hanging infra, unreleased lock and this:
> 
> ```
> time=2024-09-10T14:23:16+02:00 level=error 
> msg=terraform invocation failed in ...path... error=[...path...] 
> signal: broken pipe
> ```

For those looking to temporarily downgrade to version [0.66.9](https://github.com/Homebrew/homebrew-core/commit/9326342cbf81d8f1dcc826da2e15e41f8d28a0c8) while they fix this issue use the following: 
```shell
brew uninstall terragrunt
curl -O https://raw.githubusercontent.com/Homebrew/homebrew-core/9326342cbf81d8f1dcc826da2e15e41f8d28a0c8/Formula/t/terragrunt.rb
brew install terragrunt.rb
brew link terragrunt
brew pin terragrunt
```

**Comment by levkohimins** [comment]
Resolved in [v0.67.5](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.67.5) release.
